packages/frontend-api/src/Model/Resolver/Order/OrderResolver.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Resolver\Order;
+
+use Overblog\GraphQLBundle\Definition\Resolver\AliasedInterface;
+use Overblog\GraphQLBundle\Definition\Resolver\ResolverInterface;
+use Overblog\GraphQLBundle\Error\UserError;
+use Ramsey\Uuid\Uuid;
+use Shopsys\FrameworkBundle\Model\Customer\User\CurrentCustomerUser;
+use Shopsys\FrameworkBundle\Model\Order\Exception\OrderNotFoundException;
+use Shopsys\FrameworkBundle\Model\Order\Order;
+use Shopsys\FrameworkBundle\Model\Order\OrderFacade;
+
+class OrderResolver implements ResolverInterface, AliasedInterface
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Customer\User\CurrentCustomerUser
+     */
+    protected $currentCustomerUser;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Order\OrderFacade
+     */
+    protected $orderFacade;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CurrentCustomerUser $currentCustomerUser
+     * @param \Shopsys\FrameworkBundle\Model\Order\OrderFacade $orderFacade
+     */
+    public function __construct(
+        CurrentCustomerUser $currentCustomerUser,
+        OrderFacade $orderFacade
+    ) {
+        $this->orderFacade = $orderFacade;
+        $this->currentCustomerUser = $currentCustomerUser;
+    }
+
+    /**
+     * @param string $uuid
+     * @param string|null $urlHash
+     * @return \Shopsys\FrameworkBundle\Model\Order\Order
+     */
+    public function resolver(string $uuid, ?string $urlHash): Order
+    {
+        if (Uuid::isValid($uuid) === false) {
+            throw new UserError('Provided argument \'uuid\' is not valid.');
+        }
```

Comment:
Yeah and `urlHash` means it should be used in URL.

I disaggree with the idea it should be optional. UUID serves as an idetifier and `urlHash` is some key which validate access to it.

Reply:
We do not require uuid on frontend, so it is really weird to require it in frontend API.

Reply:
It would be really weird go against the way how API are design in general. Everything in them is provided via UUID. I do not see any reason why it should not be here.

We do not need take this legacy code into new APi method.

Reply:
Currently it is not legacy code as not registered user cannot obtain orders UUID in any way. urlHash can be at least copied from URL sent in email with new order.

Reply:
changed